paddle/fluid/operators/scatter.cu.h

```diff
@@ -28,13 +28,17 @@ using Tensor = framework::Tensor;
 template <typename T, typename IndexT = int>
 __global__ void ScatterCUDAKernel(const T* params, const IndexT* indices,
                                   T* output, size_t index_size,
-                                  size_t slice_size) {
+                                  size_t slice_size, bool overwrite) {
   CUDA_1D_KERNEL_LOOP(i, index_size * slice_size) {
     int indices_i = i / slice_size;
     int slice_i = i - indices_i * slice_size;  // offset inside the slice
     IndexT scatter_i = indices[indices_i];
     IndexT out_i = scatter_i * slice_size + slice_i;
-    *(output + out_i) = *(params + i);
+    if (overwrite) {
+      *(output + out_i) = *(params + i);
```

Comment:
This element-wisely overwriting may produce a wrong result. Assume that we assign two vectors A and B to the same position in this way,  the elements of the vector in this position may partially come from A and partial comes from B. Please consider the vector-wise assignment.